good-enough-practices-for-scientific-computing.tex

```diff
@@ -176,102 +172,55 @@ \section{Data Management}\label{sec:data}
-  is inherently interactive, but you can save the region chosen
-  as a set of boundary coordinates. If the
-  exact details of the procedure cannot easily be recorded,
-  save intermediate results for later checking.
+  \emph{Make each row an observation}: Data often comes in a wide format, because that
+  facilitated data entry or human inspection. Imagine one row per field site and then
+  columns for measurements made at each of several time points. Be prepared to gather such
+  columns into a variable of measurements, plus a new variable for time point.
 
 \item
-  \recommend{Give every record a unique key}.  In a well-organized
-  data set, it's easy to precisely refer to a specific record.
-  Using fields in the data (such as people's names or addresses) does
-  not guarantee this; using row numbers in tables does, but those
-  numbers will change as records are added, removed, or sorted.
-
-  We therefore recommend giving each record a unique, persistent key
-  that has no other purpose than identifying the
-  record\footnote{Student numbers and similar identifiers serve this
-    purpose in institutional databases.} The provision of such a key
-  also makes it much easier to link data sets together.
+  \recommend{Record all the steps used to process data}: Data manipulation is as integral
+  to your analysis as statistical modelling and inference. If you do not document this
+  step thoroughly, it impossible for you, or anyone else, to repeat the analysis.
+  
+  The best way to do this is to write scripts for \emph{every} stage of data processing.
+  This might feel frustratingly slow, but you will get faster with practice. The
+  immediate payoff will be the ease with which you can re-do data preparation when new
+  data arrives. You can also re-use data preparation logic in the future for related
+  projects.
+
+  When scripting is not feasible, it's important to clearly document every manual action
+  (what menu was used, what column was copied and pasted, what link was clicked, etc.).
+  Often you can at least capture \emph{what} action was taken, if not the complete
+  \emph{why}. For example, choosing a region of interest in an image is inherently
+  interactive, but you can save the region chosen as a set of boundary coordinates.
 
 \item
-  \recommend{Include complementary data in additional files}.  Raw data
-  frequently does not fully explain
-  itself. For example, if each well in an $8{\times}12$ microtitre
-  plate is used to study a specific gene knockout, the raw
-  data will have a \texttt{well} variable taking on values like ``A1''
-  and ``G12''. Create a supplementary file that relates the experimental
-  conditions to each variable in the raw data. Use the
-  same names and codes when variables in two
-  datasets refer to the same thing to simplify merging and lookup, and
-  to reduce errors.
-
+  \recommend{Anticipate the need to use multiple tables}: Raw data, even if tidy, is not
```

Comment:
I merged the two items about making a unique key and preparing for future joins (we use "merge and lookup" vocabulary in paper). I think it works!?